odbc_fdw.c

```diff
@@ -875,6 +888,205 @@ odbcGetTableSize(odbcFdwOptions* options, unsigned int *size)
 		SQLDisconnect(dbc);
 }
 
+static int strtoint(const char *nptr, char **endptr, int base)
+{
+  long val = strtol(nptr, endptr, base);
+  return (int) val;
+}
+
+static Oid oid_from_server_name(char *serverName)
+{
+  char *serverOidString;
+  char *ptr;
+  char sql[1024];
+  int serverOid;
+  HeapTuple tuple;
+  SPITupleTable *tuptable;
+  TupleDesc tupdesc;
+  int ret;
+
+  if ((ret = SPI_connect()) < 0) {
+    elog(ERROR, "oid_from_server_name: SPI_connect returned %d", ret);
+  }
+
+  sprintf(sql, "SELECT oid FROM pg_foreign_server where srvname = '%s'", serverName);
+  if (ret = SPI_execute(sql, true, 1) != SPI_OK_SELECT) {
+    elog(ERROR, "oid_from_server_name: Get server name from Oid query Failed, SP_exec returned %d.", ret);
+  }
+
+  if (SPI_tuptable->vals[0] != NULL)
+  {
+    tupdesc  = SPI_tuptable->tupdesc;
+    tuptable = SPI_tuptable;
+    tuple    = SPI_tuptable->vals[0];
+
+    serverOidString = SPI_getvalue(tuple, tupdesc, 1);
+    serverOid = strtoint(serverOidString, &ptr, 10);
+  } else {
+    elog(ERROR, "Foreign server %s doesn't exist", serverName);
+  }
+
+  SPI_finish();
+  return serverOid;
+}
+
+Datum
+odbc_table_size(PG_FUNCTION_ARGS)
+{
+  char *serverName = text_to_cstring(PG_GETARG_TEXT_PP(0));
+  char *tableName = text_to_cstring(PG_GETARG_TEXT_PP(1));
+  char *defname = "table";
+  int *tableSize;
```

Comment:
You're calling `odbcGetTableSize` on this uninitialized pointer and that function assigns to the pointed to object, so some random mem will be overwritten. (I know you hate functions returning values through pointer instead of using return values, buy hey)

You should just declare the int `int tableSize` and pass `&tableSize` to `odbcGetTableSize`. (and then return `PG_RETURN_INT32(tableSize);`.